CorsixTH/Lua/app.lua

```diff
@@ -374,6 +377,25 @@ function App:init()
   return true
 end
 
+--! Works out the intended location of the gamelog file.
+--!return full path gamelog should exist at
+function App:getGamelogPath()
+  local config_path = self.command_line["config-file"] or ""
+  config_path = config_path:match("^(.-)[^" .. pathsep .. "]*$")
+  return config_path .. "gamelog.txt"
+end
+
+--! Checks and creates the gamelog file if it does not exist.
+function App:initGamelogFile()
+  local gamelog_path = self:getGamelogPath()
+  local gamelog = io.open(gamelog_path, "r")
+  if gamelog and gamelog:close() then return end
```

Comment:
Should this be a generic utility function (maybe in future)?

We have `FileSystem:fileExists` but this is TH folder-centric, and not a global checker.

Reply:
Maybe it's safer to do `if gamelog then return gamelog:close() end` -- though if this generates a problem it'll unintentionally crash the game